models/nc_news.models.js

```diff
@@ -31,4 +31,27 @@ function fetchArticleById(article_id) {
     });
 }
 
-module.exports = { fetchAllTopics, fetchArticleById, fetchAllArticles };
+function fetchCommentsByArticleId(article_id) {
+  return db
+    .query(
+      `SELECT articles.article_id, comments.comment_id, comments.votes, comments.created_at, comments.author, comments.body FROM articles
+      JOIN comments
+      ON articles.article_id = comments.article_id
+      WHERE articles.article_id = $1
+      ORDER BY comments.created_at DESC;`,
+      [article_id]
+    )
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject({ status: 404, message: "Article not found." });
+      }
+      return rows;
+    });
+}
+
```

Comment:
Really nice query - this is complex so you've deciphered it clearly and presented it clearly too 🎉 

You'll need to have a look at a property other than rows.length when you add the test I've requested 🚀